### PR TITLE
De-incubate the build cache

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java [hunk 1]
```diff
@@ -37,7 +36,6 @@
  *
  * @since 3.5
  */
-@Incubating
 public class HttpBuildCache extends AbstractBuildCache {
```
Comment (lptr) [inline subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java]: `AbstractBuildCache` should not be incubating either.

**Comment by lptr** [comment]
Should we un-deprecate `WriteProperties`, too?

File: subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java (hunk 1 above)
Comment (wolfs) [reply subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java]: De-incubated.

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskIntegrationTest.groovy [hunk 2]
```diff
@@ -21,11 +21,12 @@ import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
...
 class CachedTaskIntegrationTest extends AbstractIntegrationSpec implements DirectoryBuildCacheFixture {
-    def "produces incubation warning"() {
+
+    def "does not produce incubation warning"() {
         buildFile << defineCacheableTask()
         withBuildCache().run "cacheable"
         expect:
-        result.assertOutputContains("Build cache is an incubating feature.")
+        !output.contains("Build cache is an incubating feature.")
     }
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskIntegrationTest.groovy]: Dropped.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -145,6 +145,14 @@ TestKit was first introduced in Gradle 2.6 to support developers with writing an
 
 The `CompileOptions.annotationProcessorPath` property has been promoted and is now stable.
 
+### Build cache and task output caching
+
+The [build cache](userguide/build_cache.html) and [task output caching](userguide/build_cache.html#sec:task_output_caching) were first introduced in Gradle 3.5.
+They are used in production by different teams, including the Gradle team itself, with great results.
+As of Gradle 4.6, the build cache and task output caching are no longer incubating and a public feature.
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/docs/src/docs/release/notes.md]: Yeah, you are right. Fixed.

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -145,6 +145,14 @@ TestKit was first introduced in Gradle 2.6 to support developers with writing an
...
+
+Note that the SPI to [implement your own build cache service](userguide/build_cache.html#sec:build_cache_implement) stays incubating for 4.6.
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/docs/src/docs/release/notes.md]: Dropped the version.

**Comment by wolfs** [comment]
I de-incubated `WriteProperties`, too.

**Comment by wolfs** [comment]
@lptr Please have another look.

**Pull request merged**